fido2/win_api.py

```diff
@@ -0,0 +1,730 @@
+
+    def __init__(self, handle=None):
+        self.handle = handle or ctypes.windll.user32.GetForegroundWindow()
+
+    def get_error_message(self, errno):
+        """Returns an error message given an error HRESULT value.
+
+        :param int errno: Error code from an OSError.
+        :return: An error message string.
+        :rtype: str
+
+        Example:
+            try:
+                api.make_credential(*args, **kwargs)
+            except OSError as e:
+                print(api.get_error_message(e.errno))
+        """
+
+    def make_credential(
+        self,
+        rp,
+        user,
+        public_key_cred_params,
+        client_data,
+        timeout=0,
+        resident_key=False,
+        platform_attachment=WebAuthNAuthenticatorAttachment.ANY,
+        user_verification=WebAuthNUserVerificationRequirement.ANY,
+        attestation=WebAuthNAttestationConvoyancePreference.DIRECT,
+        exclude_credentials=None,
+        extensions=None,
+    ):
+        """Make credential using Windows WebAuthN API.
+
+        :param Dict[str,Any] rp: Relying Party Entity data.
+        :param Dict[str,Any] user: User Entity data.
+        :param List[Dict[str,Any]] public_key_cred_params: List of
+            PubKeyCredentialParams data.
+        :param bytes client_data: ClientData JSON.
+        :param int timeout: (optional) Timeout value, in ms.
+        :param bool resident_key: (optional) Require resident key, default: False.
+        :param WebAuthNAuthenticatorAttachment platform_attachment: (optional)
+            Authenticator Attachment, default: any.
+        :param WebAuthNUserVerificationRequirement user_verification: (optional)
+            User Verification Requirement, default: any.
+        :param WebAuthNAttestationConvoyancePreference attestation: (optional)
+            Attestation Conveyance Preference, default: direct.
+        :param List[Dict[str,Any]] exclude_credentials: (optional) List of
+            PublicKeyCredentialDescriptor of previously registered credentials.
+        :param Any extensions: Currently not supported.
+        """
+
+        # TODO: add support for extensions
+        attestation_pointer = ctypes.POINTER(WebAuthNCredentialAttestation)()
+        WEBAUTHN.WebAuthNAuthenticatorMakeCredential(
+            self.handle,
+            ctypes.byref(WebAuthNRpEntityInformation(rp)),
+            ctypes.byref(WebAuthNUserEntityInformation(user)),
+            ctypes.byref(WebAuthNCoseCredentialParameters(public_key_cred_params)),
+            ctypes.byref(WebAuthNClientData(client_data)),
+            ctypes.byref(
+                WebAuthNMakeCredentialOptions(
+                    timeout,
+                    resident_key,
+                    platform_attachment,
+                    user_verification,
+                    attestation,
+                    exclude_credentials or [],
+                )
+            ),
+            ctypes.byref(attestation_pointer),
```

Comment:
Getting an error here:
```
Traceback (most recent call last):
  File "c:\users\cmilam\repos\platform\onica-sso\cli\onica_sso\yubikey.py", line 260, in make_fido2_credential
    algos=[i['alg'] for i in key_data['pubKeyCredParams']]
  File "c:\users\cmilam\.virtualenvs\onica-sso-testing-mmmw4rsz\lib\site-packages\fido2\client.py", line 682, in make_credential
    extensions,
  File "c:\users\cmilam\.virtualenvs\onica-sso-testing-mmmw4rsz\lib\site-packages\fido2\win_api.py", line 677, in make_credential
    ctypes.byref(attestation_pointer),
  File "_ctypes/callproc.c", line 933, in GetResult
OSError: [WinError -2146893809] Object already exists
```

Reply:
Looks to me like you might be passing an `exclude_credentials` list already containing a credential for the Authenticator you are trying to register. I do plan to wrap those OSErrors in WindowsClient and replace them with something a bit more user friendly...

Reply:
> Looks to me like you might be passing an exclude_credentials list already containing a credential for the Authenticator you are trying to register.

I can confirm your suspicion. The is the error I encountered in my testing for this case.